src/service/InvoiceService.java

```diff
@@ -0,0 +1,115 @@
+package service;
+
+import entity.Cart;
+import entity.Invoice;
+import factories.InvoiceFactory;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+
+/**
+ * This class represents methods related to invoices.
+ *
+ * @author JB
+ */
+
+public class InvoiceService {
+
+    InvoiceFactory invoiceFactory = new InvoiceFactory();
+    Cart cart = new Cart();
+
+    /**
+     * This method generates the invoice.
+     * Checks if invoice is not null and if the Customer is not null.
+     * Checks if there is any product added to the invoice.
+     * If there are, generates the invoice with order and customer details. If not then does not proceed with the invoice generation.
+     *
+     * @param invoice Invoice object on which details invoice is to be generated.
+     * @throws IllegalArgumentException If either the invoice or customer in the order is null.
+     */
+
+    public void generateInvoice(Invoice invoice) {
+        try {
+            if (invoice == null || invoice.getCustomer() == null) {
+                throw new IllegalArgumentException("The order is empty or the customer data are not added. Please check and try again.");
+            }
+            printInvoice(invoice);
+        } catch (IllegalArgumentException e) {
+            e.getMessage();
```

Comment:
nic sie z tym nie dizeje po prostu wywolujesz getMessage() -> nie jest to przypisywane do zmiennej ani wypisywane mogloby tego nie byc w ogole. 

Podejrzewam ze chciales wypisac to wypisz